src/Admin/Controllers/OrganizationsController.cs

```diff
@@ -97,6 +97,24 @@ public class OrganizationsController : Controller
         });
     }
 
+    [HttpGet]
+    public async Task<IActionResult> Create(Guid providerId)
+    {
+        var provider = await _providerRepository.GetByIdAsync(providerId);
```

Comment:
We should make sure it's a Reseller type. I think the cleanest way to do this is to add a type parameter to this sproc, so that it only returns Providers of the specified type.

This will also be an issue in existing code, which expects this query to always return an MSP type, but now it'll return an MSP or Reseller. That's what EC-433 is intended to cover, but it's worth keeping in the back of your mind. If you add this parameter to the sproc, then you could make it default to MSP type, to maintain the behaviour of existing code and make sure it doesn't accidentally start picking up Resellers as well.

Reply:
I tried that approach but then it felt weird to specify the type if we just wanted to get an entity by its `Id`. It would also have to be nullable for cases where we don't care what the Provider type is. 
I decided instead to check the type here in the controller.
I might create a new sproc `Provider_ReadByIdType` for this but I'll leave it for EC-433

Reply:
You might be right - and it might be better to be able to distinguish between "this provider ID doesn't exist" vs. "this provider is the wrong type".

In any case, this is fine for now (and always fun to use pattern matching).